pin-devices-targeting-release.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+## This script sets devices targeting a given commit to target a new commit
+## Usage: ./pin-devices-to-target-release <FULL_COMMIT_HASH> <NEW_FULL_COMMIT_HASH>
+## This is useful when you have a set of devices which are running release A,
+## have been targeted to release B, but you want them to now target release C.
+
+./check-configuration.sh || exit 1
+
+if [ -f "./resin.env" ]; then
+    source ./resin.env
+fi
+source ./balena.env
+
+COMMIT=$1
+NEW_COMMIT=$2
+if [ -z $COMMIT ]; then
+    echo "Please specify a commit to upgrade from"
+    exit 1
+elif [ -z $NEW_COMMIT ]; then
+    echo "Please specify a commit to upgrade to"
+    exit 1
+else
+    RELEASE_ID=$(./get-release-id.sh $COMMIT)
+    NEW_RELEASE_ID=$(./get-release-id.sh $NEW_COMMIT)
+fi
+echo "setting devices with commit $COMMIT with release = $RELEASE_ID to $NEW_COMMIT with release = $NEW_RELEASE_ID"
```

Comment:
While testing this, I realized that we need some guards against invalid input. Specifically
* if the source release isn't found, then the source release id that will be used is `null`, which means that a typo in the terminal could pin all unpinned devices of the app.
* the target release has the same issue, so a typo in the target release hash would unpin all devices running the specified commit.

EG:
```
./pin-devices-targeting-release.sh 9af01a00ac47cca0ed3c3bb316644f3be6eb5774 typocommit
# setting devices with commit 9af01a00ac47cca0ed3c3bb316644f3be6eb5774 with release = 985483 to typo with release = null
```
which means that all devices pinned on 9af01a00ac47cca0ed3c3bb316644f3be6eb5774 will suddenly get unpinned and update to the latest release of the app.

For the above two cases I think that we should add two checks (one for the `RELEASE_ID` and one for the `NEW_RELEASE_ID`), which would state that the script was unable to find the provided source or target release and then exit.